benchmark/infrastructure/BenchmarkMain.cpp

```diff
@@ -21,25 +22,63 @@
 #include "BenchmarkMetadata.h"
 #include "util/Algorithm.h"
 #include "util/ConfigManager/ConfigManager.h"
+#include "util/Exception.h"
 #include "util/File.h"
 #include "util/json.h"
 
 // We are not in a global header file, so its fine.
 using namespace ad_benchmark;
 
 /*
- * @brief Write the json object to the specified file.
- *
- * @param fileName The name of the file, where the json informationen
- *  should be written in.
- * @param appendToFile Should the json informationen be appended to the end
- *  of the file, or should the previous content be overwritten?
- */
-static void writeJsonToFile(nlohmann::json j, const std::string& fileName,
-                            bool appendToFile = false) {
-  ad_utility::makeOfstream(
-      fileName, appendToFile ? (std::ios::out | std::ios::app) : std::ios::out)
-      << j;
+@brief Transform the given benchmark classes and corresponding results to
+json, and write them to the specified file.
+
+@param benchmarkClassAndResults The benchmark classes togehter with their
+results of running `runAllBenchmarks`.
+@param fileName The name of the file, where the json informationen
+should be written in.
+@param appendToJsonInFile Should the json informationen be appended to the end
+of the json structure in the file, or should the previous content be
+overwritten? Note: If the json structure in the file isn't an array, an error
+will be thrown, except if the file is empty. In that case, `appendToFile` will
+be treated as `false`.
+*/
+static void writeBenchmarkClassAndBenchmarkResultsToJsonFile(
+    const std::vector<std::pair<const BenchmarkInterface*, BenchmarkResults>>&
+        benchmarkClassAndResults,
+    const std::string& fileName, bool appendToJsonInFile = false) {
+  // Convert to json.
+  nlohmann::json benchmarkClassAndBenchmarkResultsAsJson(
+      zipBenchmarkClassAndBenchmarkResultsToJson(benchmarkClassAndResults));
+  AD_CORRECTNESS_CHECK(benchmarkClassAndBenchmarkResultsAsJson.is_array());
+
+  /*
+  Add the old json arry entries to the new json array entries, if a non empty
+  file exists. Otherwise, we create/fill the file.
+  */
+  if (appendToJsonInFile && std::filesystem::exists(fileName) &&
+      !std::filesystem::is_empty(fileName)) {
+    /*
+    TODO Is there a cheaper way, than transforming the file to json, combining
+    the old and new content, and writing it to the file?
+    For example: Appending to the end of the file.
+    */
```

Comment:
I would argue that
1. it doesn't really matter in this case for performance-wise (this is local to the benchmark results class).
2. It is actually good to completely parse the JSON, validate it, and then create a new valid json and write that.
3. Of course you could check only that the file ends with a `]`, and then start at the position of the `]` write `, <NEW_JSON> ]`, but I am not sure whether this is better overall here, see 1. and 2.

So please adapt the comment to reflect that this is done deliberatly.

Reply:
Alright, I tried to improve the comment. See [fc8a672](https://github.com/ad-freiburg/qlever/pull/1155/commits/fc8a672a2e39eb3cf227e01065d0b643ddb16d3a).